src/GaussianRenderer.cpp

```diff
@@ -0,0 +1,317 @@
+    : mObjectName(std::move(objectName))
+    , mSettings(std::move(settings))
+    , mSolarSystem(std::move(solarSystem)) {
+
+  // Add to scenegraph.
+  VistaSceneGraph* pSG = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
+  mGLNode.reset(pSG->NewOpenGLNode(pSG->GetRoot(), this));
+  VistaOpenSGMaterialTools::SetSortKeyOnSubtree(
+      mGLNode.get(), static_cast<int>(cs::utils::DrawOrder::eTransparentItems));
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+GaussianRenderer::~GaussianRenderer() {
+  VistaSceneGraph* pSG = GetVistaSystem()->GetGraphicsManager()->GetSceneGraph();
+  pSG->GetRoot()->DisconnectChild(mGLNode.get());
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+void GaussianRenderer::configure(Plugin::Settings::RadianceField const& settings,
+    std::shared_ptr<Plugin::Settings>                                   pluginSettings) {
+
+  // If the ply file or the CUDA device changed, we reload everything.
+  if (mRadianceField.mPLY != settings.mPLY || mCudaDevice != pluginSettings->mCudaDevice.get()) {
+    logger().info("Loading PLY {}", settings.mPLY);
+
+    // Store the settings.
+    mRadianceField  = settings;
+    mCudaDevice     = pluginSettings->mCudaDevice.get();
+    mPluginSettings = std::move(pluginSettings);
+
+    // Select the CUDA device.
+    int num_devices;
+    CUDA_SAFE_CALL_ALWAYS(cudaGetDeviceCount(&num_devices));
+    if (mCudaDevice >= num_devices) {
+      if (num_devices == 0)
+        logger().error("No CUDA devices detected!");
+      else
+        logger().error("Provided device index exceeds number of available CUDA devices!");
+    }
+    CUDA_SAFE_CALL_ALWAYS(cudaSetDevice(mCudaDevice));
+    cudaDeviceProp prop;
+    CUDA_SAFE_CALL_ALWAYS(cudaGetDeviceProperties(&prop, mCudaDevice));
+    if (prop.major < 7) {
+      logger().error("Sorry, need at least compute capability 7.0+! (got {})", prop.major);
+    }
+
+    // Load the radiance field from the ply file.
+    std::vector<GaussianData::Pos>    pos;
+    std::vector<GaussianData::Rot>    rot;
+    std::vector<GaussianData::Scale>  scale;
+    std::vector<float>                opacity;
+    std::vector<GaussianData::SHs<3>> shs;
+
+    const int shDegree = 3;
+    Vector3f  sceneMin{};
+    Vector3f  sceneMax{};
+    mCount = loadPly<shDegree>(
+        mRadianceField.mPLY.c_str(), pos, shs, opacity, scale, rot, sceneMin, sceneMax);
+    mData = std::make_unique<GaussianData>(pos, rot, scale, opacity, shs);
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+bool GaussianRenderer::Do() {
+  auto object = mSolarSystem->getObject(mObjectName);
+  if (!object || !object->getIsBodyVisible()) {
+    return true;
+  }
+
+  cs::utils::FrameStats::ScopedTimer timer("Gaussian Renderer for " + mRadianceField.mPLY);
```

Comment:
If a windows path is provided for the ply with backslashes (like `C:\Users\xyz\...`), this will lead to errors in the FrameStats.